components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/oauth2/ApiModelToOAuthConsumerApp.java

```diff
@@ -178,4 +188,40 @@ private String getCallbackUrl(List<String> callbackURLs) {
             return null;
         }
     }
+
+    private void updateClientAuthenticationConfigurations(OAuthConsumerAppDTO appDTO,
+                                                          ClientAuthenticationConfiguration clientAuthentication) {
+
+        if (clientAuthentication != null) {
+            appDTO.setTokenEndpointAuthMethod(clientAuthentication.getTokenEndpointAuthMethod());
+            appDTO.setTokenEndpointAuthSignatureAlgorithm(clientAuthentication.getTokenEndpointAuthSigningAlg());
+            appDTO.setTlsClientAuthSubjectDN(clientAuthentication.getTlsClientAuthSubjectDn());
+        }
+    }
+
+    private void updatePARConfigurations(OAuthConsumerAppDTO appDTO, PushAuthorizationRequestConfiguration par) {
+
+        if (par != null) {
+            appDTO.setRequirePushedAuthorizationRequests(par.getRequirePushAuthorizationRequest());
+        }
+    }
+
+    private void updateRequestObjectConfiguration(OAuthConsumerAppDTO consumerAppDTO,
+                                                  RequestObjectConfiguration requestObject) {
+
+        if (requestObject != null) {
+
+            consumerAppDTO.setRequestObjectEncryptionAlgorithm(requestObject.getEncryption().getAlgorithm());
+            consumerAppDTO.setRequestObjectEncryptionMethod(requestObject.getEncryption().getMethod());
+            consumerAppDTO.setRequestObjectSignatureAlgorithm(requestObject.getRequestObjectSigningAlg());
```

Comment:
Possible NPE. Add a null check for second level objects of requestObject.

Reply:
Fixed in 1f4de9bb1b4a721922a631b048ee16397e81baea